crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/rpc.rs

```diff
@@ -31,12 +32,22 @@ pub struct UsedResources {
 impl UsedResources {
     pub fn extend(self: &mut UsedResources, other: &UsedResources) {
         self.execution_resources.vm_resources += &other.execution_resources.vm_resources;
-        self.execution_resources
-            .syscall_counter
-            .extend(&other.execution_resources.syscall_counter);
+
+        self.update_syscall_counter(&other.execution_resources.syscall_counter);
+
         self.l2_to_l1_payloads_length
             .extend(&other.l2_to_l1_payloads_length);
     }
+
+    fn update_syscall_counter(self: &mut UsedResources, syscall_counter: &SyscallCounter) {
+        for (syscall, count) in syscall_counter {
+            *self
+                .execution_resources
+                .syscall_counter
+                .entry(*syscall)
+                .or_insert(0) += count;
+        }
+    }
```

Comment:
Also i would add a test for that since our earlier tests didn't catch that

Reply:
Yup, already added: [Add failing syscalls test](https://github.com/foundry-rs/starknet-foundry/pull/1511/commits/0a30c1eda077f84344610c0fc591d9a89d8a236f)